### PR TITLE
fc for searchbarcomponent, singleteamComp, and SearchButtonComp

### PR DESCRIPTION
**Pull request by wcharles168**
Super last minute sorry alice lmao

### CODE REVIEW

File: src/components/SingleTeamComponent.tsx [hunk 1]
```diff
@@ -3,120 +3,118 @@ import { HoverWrapperComponent } from "src/components/HoverWrapperComponent";
...
-                      style={{ opacity: 0.3 }}
-                    />
-                  )}
-              &nbsp;&nbsp;&nbsp;
-            </HoverWrapperComponent>
-            </div> :
-            <div className="director">
-              <h6>{directorPerson.name}</h6>
-              <HoverWrapperComponent pp={pp} hoverDisplay={true}>
+export const SingleTeamComponent: React.FC<any> = (props) => {
```
Comment (alwu5269) [inline src/components/SingleTeamComponent.tsx:7]: Please specify the type of the props passed in. Like
`export const SingleTeamComponent: React.FC<{team: Group}> = ({team}) => {`

File: src/components/SingleTeamComponent.tsx [hunk 2]
```diff
@@ -3,120 +3,118 @@ import { HoverWrapperComponent } from "src/components/HoverWrapperComponent";
...
 
-        // for the last director, don't add the comma
-        //if (parseInt(directorIdx) < team.positionIds.length - 1) {
-        //directorsList.push(", ");
-        //}
-      }
+      // for the last director, don't add the comma
+      //if (parseInt(directorIdx) < team.positionIds.length - 1) {
+      //directorsList.push(", ");
+      //}
```
Comment (alwu5269) [inline src/components/SingleTeamComponent.tsx:60]: Please take out unnecessary/commented out code.

File: src/components/SearchbarComponent.tsx [hunk 3]
```diff
@@ -96,10 +91,10 @@ var FilteredList = createReactClass({
 
 // takes list of names from updated list and generates search buttons
 var List = createReactClass({
```
Comment (alwu5269) [inline src/components/SearchbarComponent.tsx:93]: Please make the List Component into a Functional Component

File: src/components/SearchbarComponent.tsx [hunk 4]
```diff
@@ -7,32 +7,27 @@ import * as Data from "src/data/data";
...
-  }
-  public render() {
-    return <FilteredList />;
-  }
+export const Searchbar: React.FC = () => {
+  return <FilteredList />;
 }
 
 var createReactClass = require("create-react-class");
 var FilteredList = createReactClass({
```
Comment (alwu5269) [inline src/components/SearchbarComponent.tsx:15]: Make into FC too.

File: src/components/SingleTeamComponent.tsx (hunk 2 above)
Comment (alwu5269) [reply src/components/SingleTeamComponent.tsx:60]: There is commented out code in other areas too. Please remove those too.

**Comment by alwu5269** [comment]
Also, another thing I want us to start doing is to have a specific outline/description for the PR.

PR outline:

Description: 
Explain what you did in this PR
Screenshots:
Show before and after

**Comment by mileswada** [comment]
Description of Pull Request:
Converted React Class Components into Function Components that use Hooks to maintain state.

Before:
Screen Shot 2020-07-18 at 2 58 07 PM

After:
Screen Shot 2020-07-18 at 2 58 14 PM

**Comment by alwu5269** [comment]
> Description of Pull Request:
> Converted React Class Components into Function Components that use Hooks to maintain state.
> 
> Before:
> Screen Shot 2020-07-18 at 2 58 07 PM
> 
> After:
> Screen Shot 2020-07-18 at 2 58 14 PM

Lol. Sorry I guess I should have specified what I meant by screenshots. I meant the visual changes of what you guys have done to the website. Not the screenshots of your code.

File: src/components/SearchbarComponent.tsx [hunk 5]
```diff
@@ -11,93 +11,78 @@ export const Searchbar: React.FC = () => {
...
-    // Reset search bar when there's no input text
+    // Reset search bar whe there's no input text
     if (searchBarText == "") {
-      this.setState({ items: [] });
+      setButtons([]);
     } else {
-      var updatedList = this.state.initialItems;
-      updatedList = updatedList.filter(function (item: any) {
+      setInitialState();
+      var updatedList = names.filter(function (name: string) {
```
Comment (alwu5269) [inline src/components/SearchbarComponent.tsx]: Sorry I should have said this earlier but please try and use `const`. If it has to be dynamic then `let`. Try not to use var.

File: src/components/SearchbarComponent.tsx [hunk 6]
```diff
@@ -11,93 +11,78 @@ export const Searchbar: React.FC = () => {
...
         return (
-          item.toLowerCase().search(event.target.value.toLowerCase()) !== -1
+          name.toLowerCase().search(event.target.value.toLowerCase()) !== -1
         );
       });
       var newList: any[] = [];
```
Comment (alwu5269) [inline src/components/SearchbarComponent.tsx]: Try not to use `var`.

**Comment by alwu5269** [comment]
Since you guys deleted files, there are just too many files for me to properly request changes. The last other change I would want is to remove all `var` in IndividualComponent.tsx and other files you have refactored. Once you do, it looks good to go.

**Pull request merged**